### PR TITLE
Remodel the tags hash to conform with the requirements of 'create_pro…

### PR DESCRIPTION
**Pull request by sigbjornaib**
…vision_request'.

### CODE REVIEW

**Comment by itewk** [comment]
@A-Beck or @curt-matthews can you review?

File: Automate/RedHatConsulting_Utilities/Service/Provisioning/StateMachines/Methods.class/__methods__/create_provision_requests.rb [hunk 1]
```diff
@@ -208,13 +208,47 @@ def create_provision_requests(task, requester, number_of_vms,
...
+              @handle.log(:info, "tags before removing numbered index => #{tags}") if @DEBUG
+              # Only run if there is an array in the tags hash
+              unless tags[0].nil?
+                # Create an empty hash, the processed values will go in tags_no_index
+                tags_no_index = {}
+                tags.each do |(index,tag)|
+                  # Determine if the tag_value is an array, if so, remove the array
+                  tag.each do |(tag_name, tag_value)|
+                    if tag_value.kind_of?(Array)
+                      tag_no_value_array = { tag_name => tag_value.first }
```
Comment (A-Beck) [inline Automate/RedHatConsulting_Utilities/Service/Provisioning/StateMachines/Methods.class/__methods__/create_provision_requests.rb]: What if the tag is not single valued? l.e the function tag category has both database and security?

**Comment by A-Beck** [comment]
@sigbjornaib One question about multi-valued tags.

Otherwise, looks good - thanks for fixing this!

File: Automate/RedHatConsulting_Utilities/Service/Provisioning/StateMachines/Methods.class/__methods__/create_provision_requests.rb (hunk 1 above)
Comment (sigbjornaib) [reply Automate/RedHatConsulting_Utilities/Service/Provisioning/StateMachines/Methods.class/__methods__/create_provision_requests.rb]: What does the create_provision_request support? Does it support the type of input below?

{:service=>["gen"], :function=>["db", "sec"]}

Comment (sigbjornaib) [reply Automate/RedHatConsulting_Utilities/Service/Provisioning/StateMachines/Methods.class/__methods__/create_provision_requests.rb]: To answer my own question, no it does not.
 
<AEMethod create_provision_requests>   DRb::DRbRemoteError: undefined method `downcase' for ["function"]:Array (MiqAeException::MethodNotFound)

What other options exists for multi valued tags?

Comment (A-Beck) [reply Automate/RedHatConsulting_Utilities/Service/Provisioning/StateMachines/Methods.class/__methods__/create_provision_requests.rb]: @sigbjornaib It turns out that even with lifecycle provisioning, you can only provide on value per catagory, even for multi-valued tag catagory. There does not appear to be a way to provide multiple values. My comment here is invalid, and this patch is good.

**Pull request merged**